games/Snake.js

```diff
@@ -0,0 +1,315 @@
+..........
+..........
+..........
+..........`);
+setBackground(gamebg);
+
+setSolids([ player, wall, body]);
+
+// START - PLAYER MOVEMENT CONTROLS
+
+let snake = [
+  [0, 0]
+]
+
+function placeFood() {
+    let position = [Math.floor(Math.random() * 9)+1, Math.floor(Math.random() * 7)+1];
+  
+  addSprite(position[0], position[1], food);
+
+  
+}
+
+function addScore() {
+  clearText();
+  addText("SCORE: " + score, {x: 1, y: 1});
+}
+
+function gameOver() {
+  for (let x = 0; x <= 9; x++) {
+    for (let y = 0; y <= 7; y++) {
+      addSprite(x, y, gamebg);
+    }
+  }
+  clearText();
+  addText("GAME OVER!", {x:5, y:4, color:color`2`});
+  addText("Final Score: " + score, {x: 1, y:6});
+  clearInterval(interval);
+}
+
+addScore();
+
+onInput("s", () => {
+  if (direction != "n") {
+  directionToSet = "s";
+  }
+});
+
+onInput("d", () => {
+  if (direction != "w") {
+  directionToSet = "e";
+  }
+});
+onInput("a", () => {
+  if (direction != "e") {
+  directionToSet = "w";
+  }
+});
+onInput("w", () => {
+  if (direction != "s") {
+  directionToSet = "n";
+  }
+});
+
+onInput("i", () => {
+  clearText();
+  getAll().forEach(tile => {
+    clearTile(tile.x, tile.y);
+  });
+  score = 0;
+  direction = "e";
+  snake = [
+  [0, 0]
+];
+  placeFood();
+  clearInterval(interval);
+  interval = setInterval(move, 400);
+  addScore();
+});
+
+placeFood();
+
+// END - PLAYER MOVEMENT CONTROLS
+
+function move() {
+
+  direction = directionToSet;
+  
+  if (tilesWith(food) == 0) {
+    placeFood();
+  }
+  
+  switch(direction) {
+    case "n":
+      snake.push([snake[snake.length-1][0], snake[snake.length-1][1]-1])
+      break;
+    case "s":
+      snake.push([snake[snake.length-1][0], snake[snake.length-1][1]+1])
+      break;
+    case "e":
+      snake.push([snake[snake.length-1][0]+1, snake[snake.length-1][1]])
+      break;
+    case "w":
+      snake.push([snake[snake.length-1][0]-1, snake[snake.length-1][1]])
+      break;
+  }
+  //snake.pushsnake[-1]
+  
+  if (tilesWith(player, food).length != 0 || tilesWith(body, food).length != 0) {
+    console.log(tilesWith(player).length);
+    placeFood();
+    score += 1;
+    clearTile(getFirst(food).x, getFirst(food).y);
+    addScore();
+    clearTile(snake[0][0], snake[0][1]);
+
+    clearInterval(interval);
+  interval = setInterval(move, 400 - (score*10));
+  }
+  else if (snake.length > 1) {
+    
+    let removed = snake.shift();
+    clearTile(removed[0], removed[1]);
+  }
+
+  let i = 0;
+  snake.forEach(el => {
+
+    if (i+1 < snake.length) {
+      clearTile(el[0], el[1]);
+      addSprite(el[0], el[1], body);
+  }
+  else {
+    if ((el[0] < 0 || el[0] > 9) || (el[1] < 0 || el[1] > 7)) {
+      
+      gameOver();
+    } else {
+    addSprite(el[0], el[1], player);
+    }
+  }
+    if (tilesWith(player, body).length != 0) {
+      gameOver();
+    }
+    i++;
+  })
+}
+
+let interval = setInterval(move, 400);
+
+afterInput(() => {
+  /* count the number of tiles with goals
```

Comment:
what is the reason for these commented codes?

Reply:
Ah, it seems I accidentally left in some comments from the original block pushing game. I can submit a new pull request with those unnecessary lines removed.

Reply:
@SuperBoyne

Reply:
don't need a new PR! Just `files changed` >` ...` > `edit`

Reply:
Alright, I'll do that, and add some music while I'm at it. :)